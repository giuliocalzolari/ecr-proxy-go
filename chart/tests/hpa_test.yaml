suite: test HPA
templates:
- hpa.yaml
tests:
- it: manifest should match snapshot
  chart:
    version: 1.2.3
  set:
    autoscaling:
      enabled: true
  asserts:
  - matchSnapshot: {}


- it: should disable hpa
  set:
    autoscaling:
      enabled: false
  asserts:
  - hasDocuments:
      count: 0


- it: should set minReplicas
  set:
    autoscaling:
      enabled: true
      minReplicas: 42
      maxReplicas: 43
      targetCPU: 42
  asserts:
  - equal:
      path: spec.minReplicas
      value: 42
  - equal:
      path: spec.maxReplicas
      value: 43
  - contains:
      path: spec.metrics
      content:
        resource:
          name: cpu
          target:
            averageUtilization: 42
            type: Utilization
        type: Resource




