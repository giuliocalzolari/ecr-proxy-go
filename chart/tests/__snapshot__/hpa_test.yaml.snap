manifest should match snapshot:
  1: |
    apiVersion: autoscaling/v2
    kind: HorizontalPodAutoscaler
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: ecr-proxy
        app.kubernetes.io/version: 0.0.0
        helm.sh/chart: ecr-proxy-1.2.3
      name: ecr-proxy
    spec:
      maxReplicas: 11
      metrics:
        - resource:
            name: cpu
            target:
              averageUtilization: 80
              type: Utilization
          type: Resource
      minReplicas: 1
      scaleTargetRef:
        apiVersion: apps/v1
        kind: Deployment
        name: ecr-proxy
